src/components/places/ClaimPlaceModal.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { XIcon } from "../icons/Icons";
+import { ErrorText } from "../form/ErrorText";
+import { LoadingAnimation } from "../icons/LoadingAnimation";
+import { useUserState } from "../../utils/userState";
+
+export default function ClaimPlaceModal({ title, placeId, onDismiss }) {
+  const [value, setValue] = useState("");
+  const [reason, setReason] = useState("");
+  const [showError, setShowError] = useState(null);
+  const [loading, setLoading] = React.useState(false);
+  const [user] = useUserState();
+
+  const onClick = async () => {
+    setLoading(true);
+
+    if (
+      value === "" ||
+      (value === "Other" && reason.length === 0)
+    ) {
+      setShowError(true);
+      setLoading(false);
+    } else {
+      try {
+        await axios.put(`${process.env.REACT_APP_API_URL}/api/place/claim/${placeId}`, {
```

Comment:
Following the new pattern in this pr: https://github.com/alexsaltstein/HobokenEventsWebApp/pull/108 Make a function for the server call and bring it into the server file for reusability and customizability.